pandas/core/resample.py

```diff
@@ -1142,8 +1139,15 @@ def _get_time_bins(self, ax):
                                         start=first,
                                         end=last,
                                         tz=tz,
-                                        name=ax.name,
-                                        values_present=values_present)
+                                        name=ax.name)
+
+        # GH 15549
+        values_present = isinstance(getattr(self, 'obj', None),
```

Comment:
still not clear why you need this tests for obj is Series/DataFrame. that 's all it *can* be.

Reply:
also need to check that len(binner) >= 2

Reply:
ok, I was not sure about that the obj always exists
I will do the check

Reply:
there was actually a mistake, it supposed to go start from binner[-1] not binner[-2], I fixed it anyway